src/call_frame.rs

```diff
@@ -19,28 +19,62 @@ pub struct CallFrame {
     pub aux_heap: Heap,
     pub code_page: Vec<U256>,
     pub pc: u64,
-    /// Storage for the frame using a type that implements the Storage trait.
-    /// The supported types are InMemory and RocksDB storage.
+    // TODO: Storage is more complicated than this. We probably want to abstract it into a trait
+    // to support in-memory vs on-disk storage, etc.
     pub storage: Rc<RefCell<dyn Storage>>,
-    /// Transient storage should be used for temporary storage within a transaction and then discarded.
-    pub transient_storage: InMemory,
     pub gas_left: Saturating<u32>,
+    pub transient_storage: InMemory,
+}
+
+#[derive(Debug, Clone)]
+pub struct Context {
+    pub frame: CallFrame,
+    pub near_call_frames: Vec<CallFrame>,
+}
+
+impl Context {
+    pub fn new(program_code: Vec<U256>, gas_stipend: u32) -> Self {
+        Self {
+            frame: CallFrame::new_far_call_frame(program_code, gas_stipend),
+            near_call_frames: vec![],
+        }
+    }
 }
+
 impl CallFrame {
-    pub fn new(
-        program_code: Vec<U256>,
-        gas_stipend: u32,
-        storage: Rc<RefCell<dyn Storage>>,
-    ) -> Self {
+    pub fn new_far_call_frame(program_code: Vec<U256>, gas_stipend: u32) -> Self {
         Self {
             stack: Stack::new(),
             heap: Heap::default(),
             aux_heap: Heap::default(),
             code_page: program_code,
             pc: 0,
+            storage: Rc::new(RefCell::new(InMemory::default())),
```

Comment:
Are you doing this because you will override the storage value for all the frames in the context, right? This will probably change a lot with the implementation of the `far_call` opcode but you can add a comment explaining that this value is a default.

Reply:
Since I was not implementing far calls, this was just a way of keeping what was already implemented functioning, and yes, if you use later a builder with storage, it will override it.
I'll add the comment